csrc/scheduler/mma_utils.cpp

```diff
@@ -703,11 +704,51 @@ std::unordered_set<IterDomain*> getMmaDomainSet(
   return {mma_domains.begin(), mma_domains.end()};
 }
 
+bool isConsumerAllocationInnerIDProducerAllocationInnerID(
+    TensorView* consumer) {
+  // We are only looking at the allocation domains accross
+  // the producer and consumer of a LoadStoreOp
+  auto LdStOp = consumer->definition();
+  NVF_CHECK(dynamic_cast<LoadStoreOp*>(LdStOp));
+
+  // Get all the IDs from the innermost ID of the allocation domain of
+  // the consumer to the root domain of the consumer.
+  auto vals = DependencyCheck::getAllValsBetween(
+      {consumer->getRootDomain().begin(), consumer->getRootDomain().end()},
+      {consumer->getMaybeAllocationDomain().back()});
+
+  auto ids = ir_utils::filterByType<IterDomain>(vals);
+  auto idsOnPath = std::vector<IterDomain*>(ids.begin(), ids.end());
```

Comment:
I don't think this line is needed is it? Just use `ids` instead of `idsOnPath`. Also a nit: const on all these variables.

Reply:
I sort of based it on this:

https://github.com/NVIDIA/Fuser/blob/5e0c89b289edcf52c4162713ee498ea86ce80564/csrc/ir/utils.cpp#L738-L748


  // Filter so we only have iteration domains (ignore Ints used in split)
  auto all_ids = ir_utils::filterByType<IterDomain>(all_vals);
  return std::vector<IterDomain*>(all_ids.begin(), all_ids.end());